connector/extensions/web-service/src/main/java/org/oaebudt/edc/web/service/MetadataValidator.java

```diff
@@ -0,0 +1,63 @@
+package org.oaebudt.edc.web.service;
+
+
+import java.net.URL;
+import java.util.Map;
+import java.util.regex.Pattern;
+
+public class MetadataValidator {
+
+    private static final Pattern EMAIL_REGEX = Pattern.compile(
+            "^[A-Z0-9._%+-]+@[A-Z0-9.-]+\\.[A-Z]{2,}$", Pattern.CASE_INSENSITIVE
+    );
+
+
+    public static void validate(Map<String, Object> metadata) {
+
+        validateRequiredString(metadata, "legalOrganizationName");
+        validateRequiredString(metadata, "countryOfOrganization");
+        validateRequiredString(metadata, "contactPerson");
+        validateRequiredString(metadata, "dataProcessingDescription");
+        validateRequiredString(metadata, "qualityAssuranceMeasures");
+        validateRequiredString(metadata, "dataLicensingTerms");
+
+        validateUrl(metadata, "organizationWebsite");
+        validateEmail(metadata, "contactEmail");
+
+        validateLevel(metadata, "dataAccuracyLevel");
+        validateLevel(metadata, "dataGenerationTransparencyLevel");
+        validateLevel(metadata, "dataDeliveryReliabilityLevel");
+        validateLevel(metadata, "dataFrequencyLevel");
+        validateLevel(metadata, "dataGranularityLevel");
+        validateLevel(metadata, "dataConsistencyLevel");
+    }
+
+
+    private static void validateRequiredString(Map<String, Object> map, String key) {
+        if (!map.containsKey(key) || !(map.get(key) instanceof String) || ((String) map.get(key)).trim().isEmpty()) {
+            throw new IllegalArgumentException(key + " is required and must be a non-empty string");
```

Comment:
"exception-driven flow" is not a optimal pattern, exceptions should represent exceptional cases.
Please use the `Result` pattern, in the EDC core is well established and used.
So a method like this should return a `Result` that could be `successful` or `failed`, so in the `validate` method you can "merge" the result to have a comprehensive validation result.

Same for the other methods, they should return a `Result`

Reply:
Resolved